pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -425,51 +426,56 @@ func (f *framework) RunPermitPlugins(
 	statusCode := Success
 	for _, pl := range f.permitPlugins {
 		status, d := pl.Permit(pc, pod, nodeName)
-		if !status.IsSuccess() {
-			if status.Code() == Unschedulable {
-				msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
-				klog.V(4).Infof(msg)
-				return NewStatus(status.Code(), msg)
-			}
-			if status.Code() == Wait {
-				// Use the minimum timeout duration.
-				if timeout > d {
-					timeout = d
-				}
-				statusCode = Wait
-			} else {
-				msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
-				klog.Error(msg)
-				return NewStatus(Error, msg)
+		if status.IsSuccess() {
+			continue
+		}
+
+		switch status.Code() {
+		case Unschedulable:
+			msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
+			klog.V(4).Infof(msg)
+			return NewStatus(status.Code(), msg)
+		case Wait:
+			if timeout > d {
+				timeout = d
 			}
+			statusCode = Wait
+		default:
+			msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
+			klog.Error(msg)
+			return NewStatus(Error, msg)
 		}
 	}
 
+	if statusCode != Wait {
```

Comment:
`if statusCode == Success {` might be better

Reply:
I use a `needsWait` boolean instead of `statusCode` since it's only used as an indicator.